run_devstack_integration_tests.sh

```diff
@@ -6,7 +6,7 @@ source /edx/app/edxapp/venvs/edxapp/bin/activate
 cd /edx/app/edxapp/edx-platform
 mkdir -p reports
 
-pip install -r ./requirements/edx/testing.txt
+sudo pip install -r ./requirements/edx/testing.txt
```

Comment:
Had to run the `pip install` with `sudo` because without it a package `codejail` was not getting installed. Running this command in CI with `sudo` doesn't affect anything so it's safe to use anyway.

More details can be seen in [this comment](https://github.com/mitodl/rapid-response-xblock/issues/124#issuecomment-1105154275). Al

Reply:
`sudo` will install the package systemwide, Did you check to install with `--user` option instead `sudo` here ?

Reply:
I didn't. I instead followed the edX pattern on this [As seen here](https://github.com/openedx/edx-platform/blob/master/.github/workflows/unit-tests.yml#L60). Also, even installing this system-wide for a test instance by GitHub which gets removed later on doesn't hurt much. What do you think? @asadiqbal08

Reply:
sudo will effect the installation pattern here. mean, installing package globally in your python installation for all users.

Reply:
But used based(`codejail`) package isn't affecting us in this xBlock's case, Since we are installing testing.txt from edX itself so we had to do this to make sure all the packages get installed too. edX is also following the same approach for installing packages and should refrain from following them since we are doing a similar thing, setting up edX and running tests in that.

Reply:
@asadiqbal08 if you still think this is going to affect us I'll give --user a try? LMK

Reply:
Actually, [Here](https://github.com/mitodl/rapid-response-xblock/blob/master/run_devstack_integration_tests.sh#L4) in this file we are activating the `virtualenv` and then installing the test-packages within this `virtualenv` via `pip install` but `sudo` will overwrite this concept, I believe It will now install all packages system-wide here. 
plz correct me If not be this case.

Reply:
and the thing you referenced here is pointing to github action and separate environment

Reply:
@asadiqbal08  You are correct & I agree with you about not doing this on a server. Let me define the whole scope of it.

1- It's basically done to solve the problem with `codejail` package not getting installed because pop couldn't install the package with the redirect(`edx to openedx`) work for that  (If not, You can take a look at my comment https://github.com/mitodl/rapid-response-xblock/issues/124#issuecomment-1105154275 with more details on this)
2- As things work fine with edX with the same approach (We can definitely use that as ref for something like this)
3- We are doing this in a CI/Test installation which starts runs tests and ends. (How much impact do you think it would have?)

Based on above 3, I'll try whatever you recommend? either using `--user` or letting it be.

Reply:
> and the thing you referenced [here](https://github.com/openedx/edx-platform/blob/master/.github/workflows/unit-tests.yml#L60) is pointing to github action and separate environment

FYI, We are running this script in our github actions as well. https://github.com/mitodl/rapid-response-xblock/blob/master/.github/workflows/ci.yml#L27 & in a separate env.

Reply:
Can you plz try with `--user` option  and some other [discussion](https://askubuntu.com/questions/234758/how-to-use-a-python-virtualenv-with-sudo) to follow this approach

Reply:
@asadiqbal08 I tried the --user approach and it didn't work as you can see [here](https://github.com/mitodl/rapid-response-xblock/runs/6191553757?check_suite_focus=true#step:4:148). In addition to the discussion, you mentioned above, I looked at it and I believe it trying to set up the virtualenv with sudo along with a script to run which in my opinion is already equilent to installing the packages itself with sudo as we were doing.

What would your next suggestion be?

Reply:
Referring to this discussion (that seems much relevant to us), That saying: 

> I am trying to make a python environment separate from the system one (using virtualenv) in a directory which is not under /home, because I need to use it to build other software that has to be accessible to multiple users. I can create and activate the virtualenv all right, but when I sudo some command (for example to make or compile other software), it is the system python that is used (I can tell because of the available modules). Since on Ubuntu it is not a good thing to use the root user, is there a way to tell sudo to use a virtualenv? Or maybe this is not the correct approach, and I should make a completely new python installation?


Does it not make sense to use the command like this then ? 
```
sudo /edx/app/edxapp/venvs/edxapp/bin/pip install -r ./requirements/edx/testing.txt
```

Reply:
That depends upon at which point edX itself builds the environment though plus sudo will still make it behave the same. And we'll be using our own virtual environment as mentioned above for other things like linting/formatting checks through black etc.

I can try your solution. let's see how it goes. 🤞

Reply:
Cool @asadiqbal08 it worked 👍. The checks are green now. Do you have any other reservations?